src/Executor/ReferenceExecutor.php

```diff
@@ -1012,13 +1012,13 @@ protected function completeListValue(
         $containsPromise = false;
         $completedItems = [];
         foreach ($results as $item) {
-            $fieldPath = [...$path, $i];
-            $info->path = $fieldPath;
-            $unaliasedPath = [...$unaliasedPath, $i];
```

Comment:
Because this references itself (unlike `$fieldPath`, which always uses `$path`), this caused the `$unaliasedPath` to grow with each list item.

```php
['items', 0]
['items', 0, 1]
['items', 0, 1, 2]
...
```

This resulted in quite enormous memory usage and a big performance hit. One of our large lists previously used about 175MB of memory, with the faulty version it used over 1300MB.

@ruudk This was not only a performance issue but also logically incorrect, the `$unaliasedPath` in lists was not right. Please add a test that covers this - it should fail with the previous code and work with the current code.

Reply:
Interesting! I was not expecting this at all... And it also makes me wonder: Why did PHPStan not complain about this either?

I'll create a test for it! Thanks for fixing it!

Reply:
I would not expect PHPStan to complain about this. It could be totally fine to do something like this, depending on the use case. For example, if the goal was to keep track of all visited indices of the traversed array, it could make sense.

Reply:
https://github.com/webonyx/graphql-php/pull/1580